src/components/course/data/utils.jsx

```diff
@@ -166,40 +166,47 @@ export function findCouponCodeForCourse(couponCodes, catalogList = []) {
   }));
 }
 
+export const determineIfOfferRedeemable = ({
+  remainingBalance,
+  remainingBalanceForUser,
+  remainingApplications,
+  remainingApplicationsForUser,
+  isCurrent,
+}, coursePrice) => {
+  const hasRemainingBalance = !isNil(remainingBalance) ? remainingBalance >= coursePrice : true;
+  const hasRemainingBalanceForUser = !isNil(remainingBalanceForUser) ? remainingBalanceForUser >= coursePrice : true;
+  const hasRemainingApplications = !isNil(remainingApplications) ? remainingApplications > 0 : true;
+  const hasRemainingApplicationsForUser = !isNil(remainingApplicationsForUser)
+    ? remainingApplicationsForUser > 0
+    : true;
+
+  const isOfferRedeemable = [
+    hasRemainingBalance,
+    hasRemainingBalanceForUser,
+    hasRemainingApplications,
+    hasRemainingApplicationsForUser,
+  ];
+  if (isCurrent !== undefined) {
+    const hasCurrent = !isNil(isCurrent) ? isCurrent : true;
+    isOfferRedeemable.push(hasCurrent);
+  }
+  return isOfferRedeemable;
+};
+
 /**
  * Determines whether an enterprise offer may be applied
  * to a course given the course price and its remaining spend/balance.
  *
  * @param {object} args
  * @param {object} args.offer An enterprise offer.
  * @param {number} args.coursePrice The price of the course.
- * @returns Whether the offer is redeemable for the course.
  */
-const isOfferRedeemableForCourse = ({ offer, coursePrice }) => {
-  let hasRemainingBalance = true;
-  let hasRemainingBalanceForUser = true;
-  let hasRemainingApplications = true;
-  let hasRemainingApplicationsForUser = true;
-
-  if (!isNil(offer.remainingBalance)) {
-    hasRemainingBalance = offer.remainingBalance >= coursePrice;
-  }
-  if (!isNil(offer.remainingBalanceForUser)) {
-    hasRemainingBalanceForUser = offer.remainingBalanceForUser >= coursePrice;
-  }
-  if (!isNil(offer.remainingApplications)) {
-    hasRemainingApplications = offer.remainingApplications > 0;
-  }
-  if (!isNil(offer.remainingApplicationsForUser)) {
-    hasRemainingApplicationsForUser = offer.remainingApplicationsForUser > 0;
-  }
-
-  return [
-    hasRemainingBalance,
-    hasRemainingBalanceForUser,
-    hasRemainingApplications,
-    hasRemainingApplicationsForUser,
-  ].every(value => value === true);
+export const isOfferRedeemableForCourse = ({ offer, coursePrice }) => {
+  const isOfferRedeemableConditions = determineIfOfferRedeemable(offer, coursePrice);
```

Comment:
nit: Good call to abstract this business logic out to its own helper function. That said, is the name a bit misleading? The business logic here is only setting up the conditions by which an offer is redeemable; it doesn't do the determination until like 208 below where we verify those conditions.